sdk/dotnet/Pulumi.Automation/ProjectSettings.cs

```diff
@@ -44,5 +47,57 @@ public class ProjectSettings
 
         internal static ProjectSettings Default(string name)
             => new ProjectSettings(name, new ProjectRuntime(ProjectRuntimeName.NodeJS));
+
+        internal bool IsDefault
+        {
+            get
+            {
+                return ProjectSettings.Comparer.Equals(this, ProjectSettings.Default(this.Name));
+            }
+        }
+
+        internal static IEqualityComparer<ProjectSettings> Comparer { get; } = new ProjectSettingsComparer();
+
+        private sealed class ProjectSettingsComparer : IEqualityComparer<ProjectSettings>
+        {
+            bool IEqualityComparer<ProjectSettings>.Equals(ProjectSettings? x, ProjectSettings? y)
+            {
+                if (x == null)
+                {
+                    return y == null;
+                }
+
+                if (y == null)
+                {
+                    return x == null;
+                }
+
+                return x.Name == y.Name &&
+                        ProjectRuntime.Comparer.Equals(x.Runtime, y.Runtime) &&
+                        x.Main == y.Main &&
+                        x.Description == y.Description &&
+                        x.Author == y.Author &&
+                        x.Website == y.Website &&
+                        x.License == y.License &&
+                        x.Config == y.Config &&
+                        ProjectTemplate.Comparer.Equals(x.Template, y.Template) &&
+                        ProjectBackend.Comparer.Equals(x.Backend, y.Backend);
+            }
+
+            int IEqualityComparer<ProjectSettings>.GetHashCode(ProjectSettings obj)
+            {
+                var hash = new HashCode();
+                hash.Add(obj.Name);
+                hash.Add(obj.Main);
+                hash.Add(obj.Description);
+                hash.Add(obj.Author);
+                hash.Add(obj.Website);
+                hash.Add(obj.License);
+                hash.Add(obj.Config);
+                hash.Add(obj.Backend);
+                // fields with custom Comparer skipped for efficiency
+                return hash.ToHashCode();
```

Comment:
That's fine.. omitting values from hashing is fine. I don't have concrete use cases for GetHashCode here it's just for correctness sake so GetHashCode corresponds to the same notion of equality as Equals. Just in case.. Even `return 0` is an inefficient but valid implementation.